geoh5py/shared/entity.py

```diff
@@ -168,26 +171,14 @@ def clipping_ids(self) -> list[uuid.UUID] | None:
         """
         return self._clipping_ids
 
-    def copy(self, parent=None, copy_children: bool = True, **kwargs):
+    @abstractmethod
+    def mask_by_extent(self, extent: list[float]):
```

Comment:
**issue (type-hint, inconsistency):** all implementations seem to have type `np.ndarray`